### ISSUE TITLE
Router changes escaping of request URI

### ISSUE THREAD

**Issue by miracle2k** [flynn/flynn#177]
This is a bit of a painful one. The Problem: [Thumbor](https://github.com/thumbor/thumbor/) validates requests via a signed URL. If I am running it behind Flynn, and the URL contains brackets `(, )` the signatures stop matching.

The reason I believe is [URL.RequestURI()](http://golang.org/src/pkg/net/url/url.go?s=17269:17302#L674) in go, which escapes special characters. [Request.write](http://golang.org/src/pkg/net/http/request.go?s=12173:12215#L374) uses this method when writing itself to the socket, which is in turn what [ClientConn.Write](http://golang.org/src/pkg/net/http/httputil/persist.go?s=8502:8560#L337) uses, which we call in the router.

In other words, while the incoming request contains the bytes `()`, the outgoing request to the backend encodes them.

It seems the way this is to be avoided is by setting `URL.Opaque` to the raw value. Indeed, adding `req.URL.Opaque = req.URL.Path` before writing the request fixes it for me, though I am not 100% sure about how it is supposed to interact with URL.RawQuery.

Original Thumbor Ticket for reference: https://github.com/thumbor/thumbor/issues/344
